windows/CMakeLists.txt

```diff
@@ -20,26 +21,51 @@ target_link_libraries(hidapi_winapi
     PUBLIC hidapi_include
 )
 
-set_target_properties(hidapi_winapi
-    PROPERTIES
-        EXPORT_NAME "winapi"
-        OUTPUT_NAME "hidapi"
-        VERSION ${PROJECT_VERSION}
-        PUBLIC_HEADER "${HIDAPI_PUBLIC_HEADERS}"
-)
+if(BUILD_SHARED_LIBS)
+    set_target_properties(hidapi_winapi
+        PROPERTIES
+            EXPORT_NAME "winapi"
+            OUTPUT_NAME "hidapi"
+            VERSION ${PROJECT_VERSION}
+            PUBLIC_HEADER "${HIDAPI_PUBLIC_HEADERS}"
+    )
 
-# compatibility with find_package()
-add_library(hidapi::winapi ALIAS hidapi_winapi)
-# compatibility with raw library link
-add_library(hidapi ALIAS hidapi_winapi)
-
-if(HIDAPI_INSTALL_TARGETS)
-    install(TARGETS hidapi_winapi EXPORT hidapi
-        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-        PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/hidapi"
+    # compatibility with find_package()
+    add_library(hidapi::winapi ALIAS hidapi_winapi)
+    # compatibility with raw library link
+    add_library(hidapi ALIAS hidapi_winapi)
+
+    if(HIDAPI_INSTALL_TARGETS)
+        install(TARGETS hidapi_winapi EXPORT hidapi
+            RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+            LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+            ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
+            PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/hidapi"
+        )
+    endif()
+else()
+    target_compile_definitions(hidapi_winapi PUBLIC HIDAPI_STATIC)
+    set_target_properties(hidapi_winapi
+        PROPERTIES
+            EXPORT_NAME "winapi"
+            OUTPUT_NAME "hidapi-static"
```

Comment:
this is a breaking/backward incompatible changes

I'd say that most CMake changes are backward-incompatible and not even directly related to the original issue

Reply:
This is quite correct. Without the changes all produced libraries have the same name `hidapi`, no matter what configuration (`debug`, `release`) and build mode (`static`, `shared`). This is a bit of a hindrance during installation.

Reply:
Can you share more info about your use-case, where you need to build/install both shared and static versions of the library?

Reply:
I use HIDAPI for communication with self-built Ardunio-based input devices. For these to work in Unity I need the DLL, for all other applications in C/C++ I prefer to use the static variant. Everything needs to be debugged as well, so I need both the static and the dynamic library in the Debug and Release configurations.

Reply:
For debug/non-debug I'd have to think a bit.
As for Shared/Static - looks like you're using the library in entirely different projects.
What stops you from having a separate shared and static builds under different locations?

Reply:
Of course I can also create four directories and build each configuration individually. Then, of course, I must always take care of four directories when copying and distributing between projects and computers. To have everything in one directory is much clearer. Works also with most other libraries. Especially with the configurations different names are very practical, because these are also written into the CMake-Target-Exports. So I can add HIDAPI to a project via CMake, depending on the configuration the appropriate library is used automatically. In different directories I always have to run CMake with a different path for HIDAPI when changing the configuration.

One directory with four files and e.g. the CMake targets `hidapi::hidapi` and ` hidapi::hidapi_static` makes a lot easier. You can simply add HIDAPI to the project and choose between static and shared. Debug/release is done by CMake during project generation. Having everything at once also ensures that you don't work with different versions.

Reply:
Are your projects CMake-based?